modules/core/keeper/msg_server.go

```diff
@@ -852,6 +852,21 @@ func (k Keeper) ChannelUpgradeAck(goCtx context.Context, msg *channeltypes.MsgCh
 
 	ctx.Logger().Info("channel upgrade ack succeeded", "port-id", msg.PortId, "channel-id", msg.ChannelId)
 
+	// Move channel to OPEN state if both chains have finished flushing any in-flight packets.
+	if msg.CounterpartyFlushStatus == channeltypes.FLUSHCOMPLETE {
```

Comment:
Used this instead of `channel.FlushStatus == FLUSHCOMPLETE && counterpartyChannel.FlushStatus == FLUSHCOMPLETE` so as to not consume gas unnecessarily when getting the channel.

Reply:
i guess the `msg.CounterpartyFlushStatus` has been proven with the `CounterpartyChannelProof` on the `ChannelUpgradeAck` handler so this seems fine.

Reply:
happy to go ahead with this for now, we can revisit it when doing the team walkthrough/audit.

Something about looking up the channels and stuff at the core msg server layer always seems a bit off to me but I know there's so many options here. I guess we could also do something like:

```go
if msg.CounterpartyFlushStatus == FLUSHCOMPLETE && !k.channelKeeper.HasInflightPackets() {
 // upgrade open
}
```

But anyways, we can talk about that later

Reply:
I think that should also work just fine no? Super cleaner. Will go with current approach for now but probably open a PR for ^ (need to tweak visibility of `hasInflightPackets` and just verify all tests pass correctly). Dope suggestion.